diff -Naur a/Makefile b/Makefile
--- a/Makefile	2021-03-08 07:56:18.000000000 +0000
+++ b/Makefile	2021-03-29 12:40:46.752055880 +0000
@@ -3,7 +3,7 @@
 else
 srcdir=$(shell pwd)
 endif
-CFLAGS = -O2 -g
+CFLAGS += -O2 -g
 DWZ_VERSION := $(shell cat $(srcdir)/VERSION)
 override CFLAGS += -Wall -W -D_FILE_OFFSET_BITS=64 \
 	-DDWZ_VERSION='"$(DWZ_VERSION)"' $(shell cat $(srcdir)/COPYRIGHT_YEARS)
@@ -14,7 +14,7 @@
 mandir = $(datarootdir)/man
 OBJECTS = dwz.o hashtab.o sha1.o dwarfnames.o
 dwz: $(OBJECTS)
-	$(CC) $(LDFLAGS) -o $@ $^ -lelf
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ -lelf
 install: dwz
 	install -D dwz $(DESTDIR)$(bindir)/dwz
 	install -D -m 644 $(srcdir)/dwz.1 $(DESTDIR)$(mandir)/man1/dwz.1
