diff -up dwz/Makefile.1~ dwz/Makefile
--- dwz/Makefile.1~	2022-11-10 13:02:57.496295997 +0100
+++ dwz/Makefile	2022-11-10 13:03:33.473326181 +0100
@@ -4,7 +4,7 @@ else
 srcdir=$(shell pwd)
 endif
 
-CFLAGS = -O2 -g
+CFLAGS += -O2 -g
 DWZ_VERSION := $(shell cat $(srcdir)/VERSION)
 CFLAGS_VERSION = -DDWZ_VERSION='"$(DWZ_VERSION)"'
 CFLAGS_COPYRIGHT = $(shell cat $(srcdir)/COPYRIGHT_YEARS)
@@ -30,7 +30,7 @@ ifneq "$(XXH_INLINE_ALL_WORKS)" "1"
 LIBS += -lxxhash
 endif
 dwz: $(OBJECTS)
-	$(CC) $(LDFLAGS) -o $@ $^ $(LIBS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ $(LIBS)
 args.o: native.o
 args.o: CFLAGS_FOR_SOURCE = \
 	-DNATIVE_ENDIAN_VAL=$(NATIVE_ENDIAN_VAL) \
