diff -Naur dwz-0.13/Makefile dwz-0.13.tpg/Makefile
--- dwz-0.13/Makefile	2019-08-02 07:58:18.000000000 +0000
+++ dwz-0.13.tpg/Makefile	2019-08-03 10:19:29.886053758 +0000
@@ -3,7 +3,7 @@
 else
 srcdir=$(shell pwd)
 endif
-CFLAGS = -O2 -g
+CFLAGS += -O2 -g
 DWZ_VERSION := $(shell cat $(srcdir)/VERSION)
 override CFLAGS += -Wall -W -D_FILE_OFFSET_BITS=64 -DDWZ_VERSION='"$(DWZ_VERSION)"'
 prefix = /usr
@@ -13,7 +13,7 @@
 mandir = $(datarootdir)/man
 OBJECTS = dwz.o hashtab.o sha1.o dwarfnames.o
 dwz: $(OBJECTS)
-	$(CC) $(LDFLAGS) -o $@ $^ -lelf
+	$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $^ -lelf
 install: dwz
 	install -D dwz $(DESTDIR)$(bindir)/dwz
 	install -D -m 644 $(srcdir)/dwz.1 $(DESTDIR)$(mandir)/man1/dwz.1
